### PR TITLE
CIRC-728: Declared lost/aged to lost item: renewal (effect on lost item fees) - ACTUAL COST

### PR DESCRIPTION
**Pull request by yusuf-murodov-epam**
### Purpose
Implementation part of Actual Cost feature
### Approach
Added functionality to refund/close LostItemActualCostFee when renew item, write tests for the change 
### Learning
https://issues.folio.org/browse/CIRC-728

### CODE REVIEW

File: src/test/java/api/loans/scenarios/OverrideRenewDeclaredLostItemTest.java [hunk 1]
```diff
@@ -36,4 +43,42 @@ void lostFeeCancellationDoesNotTriggerMarkingItemAsLostAndPaid() {
...
+    declareItemLostWithActualCost(itemFeeActualCost, itemProcessingFee);
+
+    assertThat(feeFineActionsClient.getAll(), hasSize(2));
+    assertThat(loan, hasLostItemFeeActualCost(isOpen(itemFeeActualCost)));
+    assertThat(loan, hasLostItemProcessingFee(isOpen(itemProcessingFee)));
+
+    feeFineAccountFixture.payLostItemActualCostFee(loan.getId(), 3.0);
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId(), 3.0);
+
+    overrideRenewalFixture.overrideRenewalByBarcode(loan, servicePointsFixture.cd1().getId());
```
Comment (roman-barannyk) [inline src/test/java/api/loans/scenarios/OverrideRenewDeclaredLostItemTest.java:60]:
```suggestion
    feeFineAccountFixture.payLostItemActualCostFee(loan.getId(), 3.0);
    feeFineAccountFixture.payLostItemProcessingFee(loan.getId(), 3.0);
    overrideRenewalFixture.overrideRenewalByBarcode(loan, servicePointsFixture.cd1().getId());
```

File: src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java [hunk 2]
```diff
@@ -351,7 +351,41 @@ protected void useChargeableRefundableLostItemFee(double itemFee, double process
...
+        processingFee, itemFee))
+      .chargeProcessingFeeWhenDeclaredLost(processingFee)
+      .withActualCost(itemFee).refundFeesWithinMinutes(1)).getId());
+
+    declareItemLost();
+    postLostItemActualCostFeeAccountForLoan(itemFee);
+  }
+
+  private void postLostItemActualCostFeeAccountForLoan(double amount) {
+
```
Comment (roman-barannyk) [inline src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java:367]: remove empty line

File: src/test/java/api/loans/scenarios/OverrideRenewDeclaredLostItemTest.java [hunk 3]
```diff
@@ -36,4 +43,42 @@ void lostFeeCancellationDoesNotTriggerMarkingItemAsLostAndPaid() {
...
+    final double itemFeeActualCost = 10.0;
+    declareItemLostWithActualCost(itemFeeActualCost, 0.0);
+
+    assertThat(feeFineActionsClient.getAll(), hasSize(1));
+    assertThat(loan, hasLostItemFeeActualCost(isOpen(itemFeeActualCost)));
+
+    feeFineAccountFixture.payLostItemActualCostFee(loan.getId(), 3.0);
+
+    overrideRenewalFixture.overrideRenewalByBarcode(loan, servicePointsFixture.cd1().getId());
+
```
Comment (roman-barannyk) [inline src/test/java/api/loans/scenarios/OverrideRenewDeclaredLostItemTest.java:79]:
```suggestion
    feeFineAccountFixture.payLostItemActualCostFee(loan.getId(), 3.0);
    overrideRenewalFixture.overrideRenewalByBarcode(loan, servicePointsFixture.cd1().getId());
```

File: src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java (hunk 2 above)
Comment (roman-barannyk) [inline src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java:367]: remove empty line

File: src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java [hunk 4]
```diff
@@ -351,7 +351,41 @@ protected void useChargeableRefundableLostItemFee(double itemFee, double process
...
+      .withName(String.format("Test lost policy processing fee %s, item fee %s",
+        processingFee, itemFee))
+      .chargeProcessingFeeWhenDeclaredLost(processingFee)
+      .withActualCost(itemFee).refundFeesWithinMinutes(1)).getId());
+
+    declareItemLost();
+    postLostItemActualCostFeeAccountForLoan(itemFee);
+  }
+
+  private void postLostItemActualCostFeeAccountForLoan(double amount) {
```
Comment (roman-barannyk) [inline src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java:366]:
```suggestion
  protected void createLostItemFeeActualCostAccount(double amount) {
```

File: src/main/java/org/folio/circulation/domain/FeeFine.java [hunk 5]
```diff
@@ -10,10 +10,11 @@
 public class FeeFine {
   public static final String OVERDUE_FINE_TYPE = "Overdue fine";
   public static final String LOST_ITEM_FEE_TYPE = "Lost item fee";
+  public static final String LOST_ITEM_FEE_ACTUAL_COST_TYPE = "Lost item fee (actual cost)";
   public static final String LOST_ITEM_PROCESSING_FEE_TYPE = "Lost item processing fee";
 
   private static final Set<String> LOST_ITEM_FEE_TYPES =
-    unmodifiableSet(hashSet(LOST_ITEM_FEE_TYPE, LOST_ITEM_PROCESSING_FEE_TYPE));
+    unmodifiableSet(hashSet(LOST_ITEM_FEE_TYPE, LOST_ITEM_FEE_ACTUAL_COST_TYPE, LOST_ITEM_PROCESSING_FEE_TYPE));
```
[the remark this answers is not in the document]
Comment (yusuf-murodov-epam) [reply src/main/java/org/folio/circulation/domain/FeeFine.java]: fixed

File: src/test/java/api/loans/scenarios/OverrideRenewDeclaredLostItemTest.java (hunk 3 above)
Comment (yusuf-murodov-epam) [reply src/test/java/api/loans/scenarios/OverrideRenewDeclaredLostItemTest.java]: fixed

File: src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java (hunk 2 above)
Comment (yusuf-murodov-epam) [reply src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java]: removed

File: src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java (hunk 4 above)
Comment (yusuf-murodov-epam) [reply src/test/java/api/loans/scenarios/RefundDeclaredLostFeesTestBase.java]: fixed

File: src/test/java/api/support/fixtures/FeeFineAccountFixture.java [hunk 6]
```diff
@@ -118,7 +125,13 @@ public IndividualResource createManualFeeForLoan(IndividualResource loan, double
...
     return accountsClient.getMany(exactMatch("loanId", loanId.toString())
-      .and(exactMatch("feeFineType", "Lost item fee")))
+        .and(exactMatch("feeFineType", "Lost item fee")))
+      .getFirst();
+  }
+
+  private JsonObject getLostItemActualCostFeeAccount(UUID loanId) {
+    return accountsClient.getMany(exactMatch("loanId", loanId.toString())
+        .and(exactMatch("feeFineType", "Lost item fee (actual cost)")))
       .getFirst();
```
[the remark this answers is not in the document]
Comment (yusuf-murodov-epam) [reply src/test/java/api/support/fixtures/FeeFineAccountFixture.java]: fixed

**Review by Mykyta-Varenyk — APPROVED** [review]

File: src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java [hunk 7]
```diff
@@ -278,7 +280,7 @@ public ReferenceDataContext(Loan loan, DeclareItemLostRequest request, String st
 
     public ReferenceDataContext withLostItemPolicy(LostItemPolicy lostItemPolicy) {
       this.lostItemPolicy = lostItemPolicy;
-      this.loan = loan.withLostItemPolicy(lostItemPolicy);
+      this. loan = loan.withLostItemPolicy(lostItemPolicy);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java:283]:
```suggestion
      this.loan = loan.withLostItemPolicy(lostItemPolicy);
```

Comment (yusuf-murodov-epam) [reply src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java:283]: this change was added to my PR by merging master incorrectly 
so, please check this : [here](https://github.com/folio-org/mod-circulation/pull/1150)

**Pull request closed without merging**